### PR TITLE
Add climate baseline verification script

### PR DESCRIPTION
**Pull request by rod-glover**
Verification script: `scripts/verify-load-climate-baseline-values.py`

This script is to be run after running the load script `scripts/load-climate-baseline-values.py`. It checks the database to make sure that the load script actually did what we wanted it to do. The verification script raises an exception where it finds any discrepancy.

### CODE REVIEW

**Comment by rod-glover** [comment]
This script may not quite be finished: I am awaiting some final information from Faron on certain data (precip values not from rainfall) that should be either loaded or filtered out of queries against the climatology datasets. This may affect the verification script.

File: pycds/climate_baseline_helpers.py [hunk 1]
```diff
@@ -177,3 +177,130 @@ def parse_line(line):
...
+    assert networks.count() == 1, 'Network "{}" not found in database'.format(pcic_climate_variable_network_name)
+    network = networks.first()
+
+    # Temperature variables
+    for (name, keyword, kwd) in [
+        ('Tx_Climatology', 'maximum', 'Max.'),
+        ('Tn_Climatology', 'minimum', 'Min.'),
+    ]:
+        temp_variable = expect_variable(name)
+        expect_variable_attr(temp_variable, 'unit', u'celsius')
```
Comment (jameshiebert) [inline pycds/climate_baseline_helpers.py]: You shouldn't need to use ``u''`` style unicode strings. Those were a vestige of the first couple minor versions of Python 3 (3.0, 3.1) that everyone realized was a mistake and never used again. E.g.

```python
$ python2.7
Python 2.7.12 (default, Nov 29 2016, 16:30:36) 
[GCC 4.9.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> u'foo' == 'foo'
True
$ python3.3
Python 3.3.5 (default, May 10 2016, 10:39:16) 
[GCC 4.9.3] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> u'foo' == 'foo'
True
```

Comment (jameshiebert) [inline pycds/climate_baseline_helpers.py]: You really could just use a dict here to eliminate a lot of redundancy:

```python
expected_attrs = {'unit': 'celcius', 'standard_name': 'air_temperature'}
for key, val in expected_attrs.items():
    expect_variable_attr(temp_variable, key, val)
```

File: scripts/verify-load-climate-baseline-values.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+    engine = create_engine(args.dsn)
+    session = sessionmaker(bind=engine)()
+
+    verify_baseline_network_and_variables(session)
+
+    def convert_temp_data(temps_in_10thsC):
```
Comment (jameshiebert) [inline scripts/verify-load-climate-baseline-values.py]: While it's somewhat obvious from reading the code, you may want to state your assumptions about the input data. "Temperatures are coded as integers in 10ths of a degree and the special integer -9999 represents a missing value". History shows that it's virtually guaranteed that these assumptions will change at some point and someone will try and run this same code with different units :D

File: scripts/verify-load-climate-baseline-values.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                # some lines from the middle (line #s 500, , 1000, , 1500, , 2000, )
+                {'station_native_id': '118401', 'values': convert_precip_data([45, 60, 63, 85, 85, 112, 116, 103, 75, 54, -9999, 53, ])},
+                {'station_native_id': '1091169', 'values': convert_precip_data([42, 30, 26, 21, 37, 49, 43, 44, 42, 51, 49, 46, ])},
+                {'station_native_id': '1166658', 'values': convert_precip_data([39, 31, 33, 34, 44, 63, 54, 42, 36, 30, 36, 46, ])},
+                {'station_native_id': '1108848', 'values': convert_precip_data([260, 232, 201, 153, 118, 96, 72, 75, 92, 204, 327, 313 ])},
+            ]
+        },
+    }
+
+    # Quis custode custodes?
```
Comment (jameshiebert) [inline scripts/verify-load-climate-baseline-values.py]: Is this French?

File: tests/conftest.py [hunk 4]
```diff
@@ -17,6 +17,27 @@
 def pytest_runtest_setup():
     logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
 
+
+@fixture(scope='session')
```
Comment (jameshiebert) [inline tests/conftest.py]: Ah, this is slick using the engine fixture at session scope and the session fixture at module scope. Good, good.

File: tests/test_climate_baseline_helpers.py [hunk 5]
```diff
@@ -45,10 +43,19 @@ def histories(stations):
...
+
+climatology_var_names = ['Tx_Climatology', 'Tn_Climatology', 'Precip_Climatology']
+
+
 def describe_get__or__create__pcic__climate__variables__network():
 
-    def test_creates_the_expected_new_network_record(empty_database_session):
-        sesh = empty_database_session
+    def test_creates_the_expected_new_network_record(session):
+        sesh = session
```
Comment (jameshiebert) [inline tests/test_climate_baseline_helpers.py]: Eh, you barely use the local variable sesh... 1-2x per test. Is it really worth the aliasing? Likewise for the next several tests.

File: pycds/climate_baseline_helpers.py (hunk 1 above)
Comment (rod-glover) [inline pycds/climate_baseline_helpers.py]: Duh. Thanks.

**Comment by rod-glover** [comment]
Ready for next review

**Comment by rod-glover** [comment]
File and variable options were an incorrect cut-paste from the loading script. We _could_ just verify one variable at a time, but that's not what I've coded and I think unnecessary.

I was wondering about the single-hyphen, two-char option: I occasionally but rarely see it done. I can remove the "c"s from each if you like.

**Comment by rod-glover** [comment]
Ready for next review.

I linted many of my files and fixed PEP8 compliance problems. You should see less red text. Unfortunately, I changed some files that were not originally involved in this particular branch. But you should be able to spot these easily.

**Pull request merged**